addons/knative/enable

```diff
@@ -4,30 +4,111 @@ set -e
 
 source $SNAP/actions/common/utils.sh
 
-echo "Enabling Knative"
-
-"$SNAP/microk8s-enable.wrapper" istio
-
+KNATIVE_VERSION="1.6.0"
+KNATIVE_FUNC_VERSION="0.25.0"
+KNATIVE_SERVING=true
+KNATIVE_EVENTING=false
 KUBECTL="$SNAP/kubectl --kubeconfig=${SNAP_DATA}/credentials/client.config"
 
-declare -a yamls=("https://github.com/knative/serving/releases/download/v0.24.0/serving-crds.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/eventing-crds.yaml"
-                  "https://github.com/knative/serving/releases/download/v0.24.0/serving-core.yaml"
-                  "https://github.com/knative/net-istio/releases/download/v0.24.0/net-istio.yaml"
-                  "https://github.com/knative/serving/releases/download/v0.24.0/serving-default-domain.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/eventing-core.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/in-memory-channel.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/mt-channel-broker.yaml"
-                 )
-
-for yaml in "${yamls[@]}"
+for i in "$@"
 do
-   $KUBECTL apply -f "$yaml"
-   sleep 3
+case $i in
+    --no-serving)
+    KNATIVE_SERVING=false
+    shift # past argument
+    ;;
+    --eventing)
+    KNATIVE_EVENTING=true
+    shift # past argument
+    ;;
+    *)
+          # unknown option
+    ;;
+esac
 done
 
+#requirement
+"$SNAP/microk8s-enable.wrapper" dns
```

Comment:
Does this also need `$SNAP/microk8s-enable.wrapper metrics-server`? I'm getting the following errors without it:

```
Jul 18 04:07:02 dev microk8s.daemon-kubelite[1776967]: I0718 04:07:02.327725 1776967 event.go:294] "Event occurred" object="knative-serving/webhook" fieldPath="" kind="HorizontalPodAutoscaler" apiVersion="autoscaling/v2" type="Warning" reason="FailedComputeMetricsReplicas" message="invalid metrics (1 invalid out of 1), first error is: failed to get cpu utilization: unable to get metrics for resource cpu: unable to fetch metrics from resource metrics API: the server could not find the requested resource (get pods.metrics.k8s.io)"
```

Reply:
Knative doesn't depend on metric-server is a warning msg that can be ignore

Reply:
Well I would say is optional, in case users wants to use HPA metrics to scale knative serving pods, so I would not add it, let the user add the metric-server addon or their own in case they want to use HPA in k8s

Reply:
I want to keep the dependencies to the minimum to be able to run on small devices